straxen/plugins/event_processing.py

```diff
@@ -733,58 +733,175 @@ def cs2_to_e(self, x):
 
 
 @export
+@strax.takes_config(
+    strax.Option(name='pre_s1_area_threshold', default=1e3,
+                 help='Only take S1s larger than this into account '
+                      'when calculating PeakShadow [PE]'),
+    strax.Option(name='pre_s2_area_threshold', default=1e4,
+                 help='Only take S2s larger than this into account '
+                      'when calculating PeakShadow [PE]'),
+    strax.Option(name='deltatime_exponent', default=-1.0,
+                 help='The exponent of delta t when calculating shadow'),
+    strax.Option(name='time_window_backward', default=int(1e9),
```

Comment:
sorry @dachengx, in [this note](https://xe1t-wiki.lngs.infn.it/doku.php?id=xenon:xenonnt:ac:prediction:shadow_refactor), shadow is defined in 3sec time_window, while here 1sec time_window is used. Which is correct?

Reply:
That note is outdated and I linked the latest version near the title, which is [this](https://xe1t-wiki.lngs.infn.it/doku.php?id=xenon:xenonnt:ac:prediction:shadow_ambience) one. The time window we are using is 1sec.

Reply:
What is the reason to change the time_window?

Reply:
The 'large' S2 rate is ~8.7 Hz in bkg runs. The possibility to find no large S2 peaks in 1 sec time window is very low(less than 0.0002). And when using 1 sec time window, PureAC validation is acceptable, which means 1 sec is adequate to describe the correlation.